### PR TITLE
FIX Sum period and sum tax for draft move

### PR DESCRIPTION
**Pull request by vrenaville**
I have modified function field (I can overloaded them with the new API, so it's coded with the old one).
in order to add in the context state all to take care of all move not only the posted one.

### CODE REVIEW

**Comment by sbidoul** [comment]
@vrenaville is that to print the standard Taxes report based on draft moves? In that case, do you think it's possible to extend the wizard to add the usual selection to decide if we want to print based on draft or posted entries as in other reports?

**Comment by vrenaville** [comment]
@sbidoul ,
Thanks for the review,
it's the final purpose but it's also to display the right result in the view form of the tax code . so it's consistent between form view et report.
it's make sense for you ?

**Comment by sbidoul** [comment]
@vrenaville I'm not sure I fully understand the implication.

If I remember correctly, the taxes report only uses posted entries.

Also, I'm not sure account_default_draft_move is the right place to change the behaviour of the taxes report computation.

**Comment by nbessi** [comment]
@sbidoul I guess it is the correct location to do this modification but I'm struggling to see the global impact of this patch

File: account_default_draft_move/account.py [hunk 1]
```diff
@@ -55,3 +56,29 @@ def button_cancel(self):
                              'SET state=%s '
                              'WHERE id IN %s', ('draft', tuple(self.ids)))
         return True
+
+
+class AccountTaxCode(osv.osv):
+    _inherit = 'account.tax.code'
+
+    def _sum_period(self, cr, uid, ids, name, args, context=None):
```
Comment (nbessi) [inline account_default_draft_move/account.py:64]: Please add a long and nice comment to explain the modifications

**Comment by sbidoul** [comment]
> @sbidoul https://github.com/sbidoul I guess it is the correct location
> to do this modification but I'm struggling to see the global impact of
> this patch
> 
> I'm struggling too :)

This change looks a bit suspicious to me, as I think using draft or posted
moves in a calculation must be under user control.

And this is a known limitation of the taxes report which uses only posted
moves (while all other reports have the usual "target moves" selection).

But perhaps this PR is unrelated to that report? (not had time to dig
deeper yet)

**Comment by lepistone** [comment]
Just to add to the fun, it looks like in odoo the year sum and the period sum don't do the same thing, after this change: 

https://github.com/odoo/odoo/commit/2146a270f34adde54a4e9b3a0fcd3c6108fca7ef#diff-bd4b5c30db1348e882ea0cd16ca41219L1541

It looks like in the year sum, draft moves are taken into account already.

**Comment by vrenaville** [comment]
After discussions, we have switch to account_move_locking instead of account_default_draft_move
so   no more need to change _sum_period and _sum_year

**Pull request closed without merging**